src/storage.rs

```diff
@@ -424,18 +531,18 @@ impl Session {
                 .iter()
                 .enumerate() // use index 
                 .find(|(_, k)| {
-                    *k == &label.clone().unwrap() // ohne dereferenzierung wird ein &&String mit &String verglichen
+                    *k == &label.clone().unwrap() // without dereferencing we compare  &&string with &string
                 })
                 .unwrap()
-                .0 // enumerate gibt tupel zurück, also index zurückgeben
+                .0 // enumerate returns tuple, so return index
         });
     
         // Add empty nodes to ensure the total number of nodes is a power of two.
         while nodes.len() < next_power_of_two {
             nodes.push(Node::initialize_leaf(false, true, Node::EMPTY_HASH.to_string(), Node::EMPTY_HASH.to_string(), Node::TAIL.to_string()));
         }
     
-        // baum erstellen und dabei alle nodes überprüfen, ob sie linkes oder rechtes kind sind
+        // create tree (re)checking all nodes if they are left or right child
```

Comment:
same as above, this comment isnt very clear, what does it mean to check if nodes are left or right child? does it mean tagging them? sorting them? not sure